examples/motoko_examples/simple-to-do/test/test.ts

```diff
@@ -0,0 +1,181 @@
+                ok: todos === expected_result
+            };
+        }
+    },
+    {
+        name: 'complete todo',
+        test: async () => {
+            const result = await todo_canister.complete_todo(2n);
+            const expected_result: ToDo[] = [
+                {
+                    id: 1n,
+                    description: FIRST_TODO_DESCRIPTION,
+                    completed: false
+                },
+                {
+                    id: 2n,
+                    description: SECOND_TODO_DESCRIPTION,
+                    completed: true
+                }
+            ];
+            const todos = await todo_canister.get_todos();
+
+            return {
+                ok:
+                    result === undefined &&
+                    equal_todo_list(todos, expected_result)
+            };
+        }
+    },
+    {
+        name: 'show completed todos',
+        test: async () => {
+            const expected_result: string = `\n___TO-DOs___\n(1) ${FIRST_TODO_DESCRIPTION}\n(2) ${SECOND_TODO_DESCRIPTION} ✔`;
+            const todos = await todo_canister.show_todos();
+            return {
+                ok: todos === expected_result
+            };
+        }
+    },
+    {
+        name: 'clear completed todos',
+        test: async () => {
+            const result = await todo_canister.clear_completed();
+            const expected_result: ToDo[] = [
+                {
+                    id: 1n,
+                    description: FIRST_TODO_DESCRIPTION,
+                    completed: false
+                }
+            ];
+            const todos = await todo_canister.get_todos();
+
+            return {
+                ok:
+                    result === undefined &&
+                    equal_todo_list(todos, expected_result)
+            };
+        }
+    },
+    {
+        name: 'complete todo',
+        test: async () => {
+            const result = await todo_canister.complete_todo(1n);
+            const expected_result: ToDo[] = [
+                {
+                    id: 1n,
+                    description: FIRST_TODO_DESCRIPTION,
+                    completed: true
+                }
+            ];
+            const todos = await todo_canister.get_todos();
+
+            return {
+                ok:
+                    result === undefined &&
+                    equal_todo_list(todos, expected_result)
+            };
+        }
+    },
+    {
+        name: 'clear completed todos',
+        test: async () => {
+            const result = await todo_canister.clear_completed();
+            const todos = await todo_canister.get_todos();
+
+            return {
+                ok: result === undefined && todos.length === 0
+            };
+        }
+    }
+];
+
+run_tests(tests);
+
+function equal_todo_list(list_a: ToDo[], list_b: ToDo[]): boolean {
+    if (list_a.length !== list_b.length) {
+        return false;
+    }
+    for (let i = 0; i < list_a.length; i++) {
```

Comment:
Let's not use a for loop here, use one of reduce or every or one of the other functional array methods: https://github.com/demergent-labs/azle/blob/main/contributing/coding-guidelines.md#basic-principles